### PR TITLE
Upgrade OTC to 0.16.0-sumo

### PR DESCRIPTION
**Pull request by mat-rumian**
###### Description

Fill in your description here.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by pmalek-sumo** [comment]
@mat-rumian Please fix tests :)

**Comment by mat-rumian** [comment]
@pmm-sumo please merge when you decide it's a proper time for that :)

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -2102,15 +2102,36 @@ otelcol:
...
-        ## Number of workers that dequeue batches
-        num_workers: 16
-        ## Maximum number of batches kept in memory before data is dropped
-        queue_size: 10000
-        ## Whether to retry on failure or give up and drop
-        retry_on_failure: true
+      ## Smart cascading filtering rules with preset limits.
+      cascading_filter:
+        ## (default = 30s): Wait time since the first span of a trace before making
+        ## a filtering decision
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:2108]: Not sure if we want those comments about the defaults.

If we really want it then I'd put it at the end of the comment like so (and small suggestion regarding wording used):

```suggestion
        ## Wait time between obtaining first span of a trace and making a filtering decision
        ## Default: 30s
```

**Comment by pmm-sumo** [comment]
@perk-sumo customer are asking for this capability, can we include it in 2.0.0?

**Review by pmm-sumo — APPROVED** [review]

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:2108]: I would skip the defaults too
@mat-rumian

Comment (pmm-sumo) [reply deploy/helm/sumologic/values.yaml:2108]: @sumo-drosiek I am going to update this section a little bit

Comment (pmm-sumo) [reply deploy/helm/sumologic/values.yaml:2109]: I gave it a though and I think this is complex enough and almost always requires assigning values by user to better have too much rather too little description

**Comment by pmm-sumo** [comment]
I have tested it and it now looks good

One caveat is that the default gRPC port has changed from `55680` to `4317`, though in the migration period OTC listens on both of them

**Review by perk-sumo — APPROVED** [review]

**Pull request merged**